benchmarks/dataset/flair/numpy.py

```diff
@@ -37,8 +41,21 @@ def make_federated_dataset(
         Federated dataset from the HDF5 data file.
     """
     num_classes = len(get_label_mapping(hdf5_path, use_fine_grained_labels))
-    user_num_images = get_user_num_images(hdf5_path, partition)
-    user_ids = sorted(user_num_images.keys())
+    user_id_to_weight = {
+        k: min(v, max_num_user_images)
+        for k, v in get_user_num_images(hdf5_path, partition).items()
+    }
+    median_datapoints = np.median(list(user_id_to_weight.values()))
+    base_value = scheduling_base_weight_multiplier * median_datapoints
```

Comment:
This feels like new functionality which should be abstracted out of this function `make_federated_dataset`.

Reply:
Right, if you think this (scaled median) can generally benefits most datasets. But if we only have evidence on FLAIR, I think it is ok to leave it here for now.